dll/cpl/intl/languages.c

```diff
@@ -82,32 +112,46 @@ LanguagesPageProc(HWND hwndDlg,
             break;
 
         case WM_NOTIFY:
-            if (((LPNMHDR)lParam)->code == (UINT)PSN_APPLY)
+            if (((LPNMHDR)lParam)->code == (UINT)PSN_APPLY) /* Apply changes */
             {
-                /* Apply changes */
-                if (IsDlgButtonChecked(hwndDlg, IDC_INST_FILES_FOR_ASIAN) == BST_CHECKED)
+                /* EAST ASIAN specific */
+                switch (PRIMARYLANGID(GetUserDefaultLangID()))
                 {
-                    /* EAST ASIAN specific */
-                    switch (PRIMARYLANGID(GetUserDefaultLangID()))
-                    {
-                        case LANG_JAPANESE:
-                            if (HasJapaneseIME())
+                    case LANG_JAPANESE:
+                        if (IsDlgButtonChecked(hwndDlg, IDC_INST_FILES_FOR_ASIAN) == BST_CHECKED)
+                        {
+                            if (!HasJapanesePackage(hwndDlg))
                             {
-                                EnableWindow(GetDlgItem(hwndDlg, IDC_INST_FILES_FOR_ASIAN), FALSE);
-                                CheckDlgButton(hwndDlg, IDC_INST_FILES_FOR_ASIAN, BST_CHECKED);
+                                /* Install now */
+                                ShellExecuteW(hwndDlg, NULL, L"rapps.exe", L"/INSTALL jpn-package",
+                                              NULL, SW_SHOWNORMAL);
                             }
-                            else
+                        }
+                        else
+                        {
+                            WCHAR szUninstall[MAX_PATH];
+                            if (GetJapaneseUninstallCmdLine(hwndDlg, szUninstall, _countof(szUninstall)))
                             {
-                                ShellExecuteW(hwndDlg, NULL, L"rapps.exe", L"/INSTALL mzimeja",
-                                              NULL, SW_SHOWNORMAL);
+                                /* Go to arguments of command line */
+                                PWCHAR pchArgs = PathGetArgsW(szUninstall);
+                                if (pchArgs && *pchArgs)
+                                {
+                                    --pchArgs;
+                                    *pchArgs = UNICODE_NULL; /* Cut the C string */
+                                    ++pchArgs;
```

Comment:
```suggestion
                                    /* pchArgs pointer is inside szUninstall,
                                     * so we can cut the C string this way. */
                                    --pchArgs;
                                    *pchArgs = UNICODE_NULL;
                                    ++pchArgs;
```

Reply:
"so we can cut the C string this way" --> "so we have to split both strings"  :thinking:

Reply:
Yep, this sounds better. Updated my suggestion

Reply:
Fixed.